src/components/filesystem/index.js

```diff
@@ -887,6 +919,144 @@ module.exports = {
             this.showFeedbackForm = !this.showFeedbackForm;
         },
 
+        popConversation: function(msgId) {
+            if (msgId != null) {
+                for (var i=0; i < this.conversationMonitors.length; i++ ) {
+                    let currentMessage = this.conversationMonitors[i];
+                    if(currentMessage.id == msgId) {
+                        this.conversationMonitors.splice(i, 1);
+                        break;
+                    }
+                }
+            }
+        },
+        getMessage: function(msgId) {
+            if (msgId != null) {
+                //linear scan
+                for (var i=0; i < this.messageMonitors.length; i++ ) {
+                    let currentMessage = this.messageMonitors[i];
+                    if(currentMessage.id == msgId) {
+                        return this.messageMonitors[i];
+                    }
+                }
+            }
+            return null;
+        },
+        sendFeedback: function(contents) {
+            this.showSpinner = true;
+            let that = this;
+            var trimmedContents = contents.length > 1000 ? contents.substring(0, 1000) : contents;
+            this.context.sendFeedback(trimmedContents)
+                .thenApply(function(res) {
+                    that.showSpinner = false;
+                    if (res) {
+                        console.log("Feedback submitted!");
+                        that.closeFeedbackForm(null, false);
+                    } else {
+                        that.errorTitle = 'Error sending feedback';
+                        that.errorBody = "";
+                        that.showError = true;
+                    }
+            }).exceptionally(function(throwable) {
+                that.errorTitle = 'Error sending feedback';
+                that.errorBody = throwable.getMessage();
+                that.showError = true;
+                that.showSpinner = false;
+            });
+        },
+
+        sendMessage: function(msgId, contents) {
+            let that = this;
+            let message = this.getMessage(msgId);
+            if (message != null) {
+                this.showSpinner = true;
+                var trimmedContents = contents.length > 1000 ? contents.substring(0, 1000) : contents;
```

Comment:
we define max length in the ServerMessage class. Can we reference that directly? Does it need a jsannotation?

Reply:
done